core/src/com/unciv/models/metadata/GameSettings.kt

```diff
@@ -68,4 +74,64 @@ class GameSettings {
         if (tutorialTasksCompleted.add(tutorialTask))
             save()
     }
+
+    fun updateLocaleFromLanguage() {
+        val bannedCharacters = listOf(' ', '_', '-', '(', ')') // Things not to have in enum names
+        val languageName = language.filterNot { it in bannedCharacters }
+        val code = try {
+            LocaleCode.valueOf(languageName)
+        } catch (e: IllegalArgumentException) {
+            LocaleCode.English
+        }
+        locale = Locale(code.language, code.country)
+    }
+
+    fun getCurrentLocale(): Locale {
```

Comment:
as most (if not all?) calls chain a get Collator instance - centralize that here too??